### PR TITLE
Bug 1689066 - Bad representation for deleted signatures in email notification

### PR DESCRIPTION
**Pull request by beatrice-acasandrei**

### CODE REVIEW

File: treeherder/model/data_cycling/signature_remover.py [hunk 1]
```diff
@@ -124,9 +124,9 @@ def __extract_properties(signatures) -> List[dict]:
         properties = []
```
Comment (ionutgoldan) [inline treeherder/model/data_cycling/signature_remover.py:124]: Could you add the type annotation for this method' s parameter?

File: treeherder/model/data_cycling/signature_remover.py [hunk 2]
```diff
@@ -124,9 +124,9 @@ def __extract_properties(signatures) -> List[dict]:
...
         for signature in signatures:
             signature_properties = {
-                "repository": signature.repository,
-                "framework": signature.framework,
-                "platform": signature.platform,
+                "repository": signature.repository.name,
```
Comment (ionutgoldan) [inline treeherder/model/data_cycling/signature_remover.py:127]: Could we add a test for the email' s content? So we don't have to send a real email to check this works as expected.

File: treeherder/model/data_cycling/email_service.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    @property
+    def address(self):
+        return self._address
+
+    @property
+    def subject(self):
+        return self._subject
+
+    @property
+    def content(self):
```
Comment (ionutgoldan) [inline treeherder/model/data_cycling/email_service.py:38]: Let's provide test coverage for the content itself.

File: tests/model/cycle_data/test_perfherder_cycling.py [hunk 4]
```diff
@@ -283,6 +283,20 @@ def test_signature_remover(
     assert PerformanceSignature.objects.first() == test_perf_signature
 
 
+def test_email_payload(test_perf_signature):
```
Comment (ionutgoldan) [inline tests/model/cycle_data/test_perfherder_cycling.py:286]: This feels like we're testing a private method, which isn't good practice.

**Review by ionutgoldan — CHANGES_REQUESTED** [review]
Please also rebase on latest master.

**Review by ionutgoldan — APPROVED** [review]
Looking good!

**Review by sarah-clements — APPROVED** [review]

**Pull request merged**